app/astria/prompt-webhook/route.ts

```diff
@@ -117,7 +117,10 @@ export async function POST(request: Request) {
   try {
     // Here we join all of the arrays into one.
     const allHeadshots = prompt.images;
-    const modelId = prompt.tune_id;
+    // const modelId = prompt.tune_id // model id fory tune api
+
+    const regex = /<[^>]*:(\d+):/;
```

Comment:
the modelId is created when initially calling the create request `POST /p/corporate-headshots/tunes` and saved in about line 218

Reply:
I will add a webhook that fires when all prompt images finish processing so it's easier

Reply:
> the modelId is created when initially calling the create request `POST /p/corporate-headshots/tunes` and saved in about line 218
> It is then saved using the code
> 
> ```
>     const { error: modelError, data } = await supabase
>       .from("models")
>       .insert({
>         modelId: tune.id, // store tune Id field to retrieve workflow object if needed later
>         user_id: user.id,
>         name,
>         type,
>       })
>       .select("id")
>       .single();```

Yes am aware and this particular modelid created is stored in the prompt text for packs api and not tune_id key that is why i had to use regex to extract it.

the modelId is created when initially calling the create request `POST /p/corporate-headshots/tunes`  is different from the on sent to the webhook prompt tune_id for pack api but for tune api they are the same.

Reply:
That means this should be done before the create tune post request to generate the uuid and pass it as a webhook param and update modelId from the create tune response later.

```
const { error: modelError, data } = await supabase
      .from("models")
      .insert({
        modelId: tune.id, // store tune Id field to retrieve workflow object if needed later
        user_id: user.id,
        name,
        type,
      })
      .select("id")
      .single();```
```

Then i use the uuid from the callback param instead of the modelId am i right.

I believe that would work better and be consistent.

Reply:
sounds good. Please make sure to try {} catch the request, and if fails, delete the created model?

Reply:
Alright on it.